release/ray_release/reporter/artifacts.py

```diff
@@ -27,9 +29,20 @@ def report_result(self, test: Test, result: Result):
             json.dump(test, fp, sort_keys=True, indent=4)
 
         result_file = os.path.join(self.artifacts_dir, ARTIFACT_RESULT_FILE)
+        result_dict = result.__dict__
+        metrics_dict = result_dict.pop("prometheus_metrics")
         with open(result_file, "wt") as fp:
-            json.dump(result.__dict__, fp, sort_keys=True, indent=4)
+            json.dump(result_dict, fp, sort_keys=True, indent=4)
 
         logger.info(
             f"Wrote test config and result to artifacts directory: {self.artifacts_dir}"
         )
+
+        if metrics_dict:
+            metrics_file = os.path.join(self.artifacts_dir, METRICS_RESULT_FILE)
+            with gzip.open(metrics_file, "w") as f:
+                f.write(
```

Comment:
Let's keep it consistent with the other write instruction

```suggestion
            with gzip.open(metrics_file, "wt") as fp:
                fp.write(
```